components/auth/LoginForm.js

```diff
@@ -35,137 +49,131 @@ const fields = {
   },
   passwordAgain: {
+    isSignUpField: true,
     icon: 'unlock-alt',
-    inputType: 'password',
-    onlyOnSignUp: true,
     validator: ({ signUp, password, passwordAgain }) =>
       signUp &&
       (required(passwordAgain) || isEqual(password, passwordAgain, 'auth.passwordsNotEqual')),
-    successMessage: 'auth.passwordsEqual',
+    successText: 'auth.passwordsEqual',
+    type: 'password',
   },
 };
 
-const errorValidator = values => {
+const loginFieldsKeys = Object.keys(loginFields);
+
+const loginValidator = values => {
   const errors = {};
-  Object.keys(fields).forEach(key => {
-    if (fields[key] && fields[key].validator) {
-      errors[key] = fields[key].validator(values);
+  loginFieldsKeys.forEach(key => {
+    if (loginFields[key] && loginFields[key].validator) {
+      const error = loginFields[key].validator(values);
+      if (error) {
+        errors[key] = error;
+      }
     }
   });
   return errors;
 };
 
-const keys = Object.keys(fields);
-const errorsPropsTypes = {};
-keys.forEach(field => {
-  errorsPropsTypes[field] = PropTypes.string;
-});
-
-const LoginForm = ({ onSubmit, pending, errors, allowSignUp }) => {
-  let serverErrors = { ...errors };
-
-  const handleModeSwitch = (formApi, value) => {
-    const clearTouch = {};
-    formApi.setValue('signUp', value);
-    Object.keys(fields).forEach(field => {
-      if (!formApi.values[field]) {
-        clearTouch[field] = false;
-      }
-    });
-    formApi.setAllTouched(clearTouch);
-  };
-
-  return (
-    <div>
-      <h1 className="title is-size-5 has-text-centered">
-        <Text id="auth.signIn" />
-      </h1>
-      <Form onSubmit={onSubmit} validateError={errorValidator}>
-        {formApi => {
-          // FIXME(@drapegnik): find more clean way to handle server-side errors
-          if (serverErrors) {
-            Object.keys(fields).forEach(key => {
-              formApi.setError(key, serverErrors[key]);
-            });
-            serverErrors = null;
-          }
-          return (
-            <form onSubmit={formApi.submitForm}>
-              {allowSignUp && (
-                <div className="field">
-                  <div className="control has-text-centered">
-                    <label htmlFor="signUp" className="checkbox">
-                      <Checkbox
-                        id="signUp"
-                        field="signUp"
-                        onChange={handleModeSwitch.bind(null, formApi)}
-                      />
-                      <Text id="auth.noAccount" />
-                    </label>
-                  </div>
-                </div>
-              )}
-              {!allowSignUp && (
-                <div className="signup-not-available">
-                  <Text id="auth.signup-not-available" />
-                </div>
-              )}
-              {keys.filter(key => formApi.values.signUp || !fields[key].onlyOnSignUp).map(key => (
+const LoginForm = ({ url, allowSignUp, pending }) => (
```

Comment:
please omit `pending`, remove it from reducer, and use `Formik` internal state `isSubmitting` for handling this logic